### PR TITLE
Examples installer & self-check v1

### PR DESCRIPTION
**Pull request by DoriftoShoes**
This adds an installs script for the examples pack and fixes a path issue on the tests install script.  A self-check script was also added.  The self-check script runs all of the tests in the tests pack.

Sample output of self-check:

```
root@st2stage201:/home/phool# ./self-check 
Attempting Test tests.test_packs_pack...OK!
Attempting Test tests.test_quickstart...OK!
Attempting Test tests.test_quickstart_key...OK!
Attempting Test tests.test_quickstart_rules...OK!
SUCCESS: All tests passed!
```

### CODE REVIEW

**Comment by dzimine** [comment]
LGTM! 
Style hint - one liner of what script doing in the first line - just a notch better than a name.

File: tools/self-check [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+USER='testu'
+PASS='testp'
```
Comment (dzimine) [inline tools/self-check]: pass as args? with defaults?

File: tools/self-check [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+INSTALL=`${PYTHONPACK}/st2common/bin/st2-setup-examples`
+
+EXITCODE=$?
+echo $INSTALL
+
+ST2_AUTH_TOKEN=`st2 auth ${USER} -p ${PASS} | grep token | awk '{ print $4 }'`
+export ST2_AUTH_TOKEN=${ST2_AUTH_TOKEN}
+
+TEST_ACTION_LIST=`st2 action list --pack=tests | awk '{ print $2 }' | grep -v "|" | grep -v "ref"`
```
Comment (dzimine) [inline tools/self-check]: I would bring this out to a separate script. This way I can install tests and examples once, and run them many times, when needed (assume already authenticated shell).

Comment (DoriftoShoes) [inline tools/self-check]: I don't mind assuming the token exists in the env already.  That way I can just rip out the whole bit with the user/pass.  I'll keep the install bits in but wrap them in some logic to see if they are already installed.

**Comment by dzimine** [comment]
Isn't it lovely? LGTM!

**Pull request merged**